File: Arrays/2971. Find Polygon With the Largest Perimeter.py
class Solution(object):
    def largestPerimeter(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        sum_value = 0
        perimeter = -1

        for num in nums:
            # check if the coming edge length is less
            # than the sum of the other sides
            # add that to the perimeter
            if num < sum_value:
                perimeter = num + sum_value
            sum_value += num

        return perimeter
        

File: Arrays/2971. Find Polygon With the Largest Perimeter.cpp
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        std::sort(nums.begin(), nums.end());
        long sum = 0;
        long perimeter = -1;

        for (int i = 0; i < nums.size(); i++) {
            // check if the coming edge length is less
            // than the sum of the other sides
            // add that to the perimeter
            if (nums[i] < sum) {
                perimeter = nums[i] + sum;
            }
            sum += nums[i];
        }

        return perimeter;
    }
};
